### PR TITLE
Story/cite 147

### PR DESCRIPTION
**Pull request by MaulikLim**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
(you can copy the ticket if it hasn't changed) 

[CITE-147] - Attachments need to be shown on citation page  
## Anything else the reviewer needs to know?

I have removed unused imports from MongoTemplate class.

[CITE-147]: https://diging.atlassian.net/browse/CITE-147

### CODE REVIEW

**Comment by jdamerow** [comment]
please update citesphere-model version and reopen.

**Review by jdamerow — CHANGES_REQUESTED** [review]
and then reopen when the module prs have been merged and this can be deployed.

**Comment by jdamerow** [comment]
> and then reopen when the module prs have been merged and this can be deployed.

Are the module PRs merged? If so, the versions are not updated and this is still draft PR. Is there a reason for that?

**Comment by MaulikLim** [comment]
The module PRs have been merged. I believe the spring-social-zotero 0.9 and citsphere.model 1.10 does not contain the new code and so I am using the current snapshots from develop for these modules. I also tested it on my machine. Do I need to change anything else? I kept it as draft PR as I thought it has not yet been reviewed due to module version mismatch.

**Comment by jdamerow** [comment]
I looked it over but need to see it deployed. So just reopen when the other plugins have been released and the above change has been made.

**Comment by jdamerow** [comment]
Also, which pull request is this one waiting for? for spring-social-zotero there only seem to be a PR for 148 and there is none for citesphere-model.

**Comment by MaulikLim** [comment]
Module PRs have already been merged

**Comment by MaulikLim** [comment]
Okay, I'll reopen after the plugin releases are done.

**Comment by jdamerow** [comment]
citesphere-model has already been released I believe. I'll release spring-social-zotero now.

**Comment by MaulikLim** [comment]
This was merged after the citesphere-model 1.10 release

**Comment by jdamerow** [comment]
there is a 1.11 release
https://github.com/diging/citesphere-model/releases/tag/1.11-4-g8d4b95a

**Comment by MaulikLim** [comment]
It was merged after 1.11 too.

**Comment by jdamerow** [comment]
ah right, the pull request didn't show up first in the list. I'll release it now.

**Comment by MaulikLim** [comment]
Updated the plugin versions.

File: citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java [hunk 1]
```diff
@@ -118,14 +127,23 @@ public ICitation getCitation(IUser user, String groupId, String key)
...
-    public ICitation getCitation(String key) {
-        Optional<ICitation> optional = citationStore.findById(key);
-        if (optional.isPresent()) {
-            return optional.get();
+    public List<ICitation> getAttachments(IUser user, String groupId, String key)
+            throws GroupDoesNotExistException, CannotFindCitationException, ZoteroHttpStatusException {
+        ICitationGroup group = groupManager.getGroup(user, groupId);
+        if (group != null && group.getGroupId() == new Long(groupId)) {
+            if (!group.getUsers().contains(user.getUsername())) {
+                if (!group.getUsers().contains(user.getUsername())) {
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java:135]: is there a reason for this duplicate if statement?

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
Beside that, it should be good to go (although there seems to be an issue with the syncing but I'll check that later and will write a separate ticket if necessary).

**Comment by jdamerow** [comment]
Oh and one thing, for the attachment list, let's use an attachment icon instead of a bullet point.

**Comment by MaulikLim** [comment]
Used attachment icon and removed the duplicate if condition

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**